src/hotspot/share/opto/c2compiler.cpp

```diff
@@ -469,6 +469,11 @@ bool C2Compiler::is_intrinsic_supported(const methodHandle& method, bool is_virt
   case vmIntrinsics::_fsignum:
     if (!Matcher::match_rule_supported(Op_SignumF)) return false;
     break;
+#ifdef _LP64
+  case vmIntrinsics::_PhantomReference_refersTo0:
+    if (UseCompressedOops && UseShenandoahGC) return false;
+#endif
+    break;
```

Comment:
Is this intended to disable the intrinsic on all non-64-bit platforms? Is that only for Shenandoah 64-bit? I wonder if it should just be:

```
  case vmIntrinsics::_PhantomReference_refersTo0:
    if (UseCompressedOops && UseShenandoahGC) return false;
    break;
```

Reply:
Considering `UseCompressedOops` doesn't make much sense in 32-bit mode and is set to `false`, it seems `#ifdef` can be just dropped.

Reply:
You are right. I thought flag UseCompressedOops is defined only in 64-bit  VM.
@shipilev, #ifdef was placed incorrectly - it should be after `case:`. But as you both pointed, it is not needed. I will remove it.